### DISCUSSION TITLE
allow mark any page as api router

### DISCUSSION THREAD

**Discussion by bluelovers** [vercel/next.js#13344] [Ideas]
# Feature request

## Is your feature request related to a problem? Please describe.

if we want `api` not inside `/page/api` folder or mixin `page` and `api` in same folder

we need use `custom server`
https://nextjs.org/docs/advanced-features/custom-server

i hope can more easy control it, without config `next.config` and `custom server`

## Describe the solution you'd like

export `enableApiRouter` in any file
will mark this file is `api router`

`page/my_api.ts`
```ts
export const enableApiRouter = true;

export default (req, res) => {
  res.statusCode = 200
  res.setHeader('Content-Type', 'application/json')
  res.end(JSON.stringify({ name: 'John Doe' }))
}
```

**Comment by mmmeff** [comment]
+1 on this - my team needs to define its API routes behind a custom path to avoid conflicts with other pages. Having a way to configure this would be very helpful.

**Comment by ivanpetrushev** [comment]
+1 on this - we have independent 3rd party service that is requesting our api on another URL. Imagine 'https://our-domain.com/api-v1/stuff/method1/param1". We don't have access on how they are trying to access us, but we have to be compatible with them.

One possible solution will be to use rewrites. But docs say rewrites are only available in Node.js environment and we are targeting serverless buid.
